encoders/src/main/scala/org/apache/spark/sql/execution/columnar/SharedExternalStoreUtils.scala

```diff
@@ -0,0 +1,183 @@
+    useLocatorURL = useLocatorUrl(hostList)
+    createConnection(connectionProperties, hostList)
+  }
+
+  private def createConnection(connProperties: ConnectionProperties,
+      hostList: ArrayBuffer[(String, String)]): Connection = {
+    // TODO CHECK THIS SOCKETCREATOR DEPENDENCY RESOLUTION
+    // val localhost = SocketCreator.getLocalHost
+    var index = -1
+
+    val jdbcUrl = if (useLocatorURL) {
+      connProperties.url
+    } else {
+      // TODO NEEDS TO BE HANDLED=
+      // if (index < 0) index = hostList.indexWhere(_._1.contains(localhost.getHostAddress))
+      if (index < 0) index = Random.nextInt(hostList.size)
+      hostList(index)._2
+    }
+
+    // TODO : REMOVE THIS AND ADD DEPENDENCY TO ACCESS ClientAttribute.THRIFT_LOB_DIRECT_BUFFERS
+    /**
+     * Use direct ByteBuffers when reading BLOBs. This will provide higher
+     * performance avoiding a copy but caller must take care to free the BLOB
+     * after use else cleanup may happen only in a GC cycle which may be delayed
+     * due to no particular GC pressure due to direct ByteBuffers.
+     */
+    val THRIFT_LOB_DIRECT_BUFFERS = "lob-direct-buffers"
+    // -
+
+    // enable direct ByteBuffers for best performance
+    val executorProps = connProperties.executorConnProps
+    // executorProps.setProperty(ClientAttribute.THRIFT_LOB_DIRECT_BUFFERS, "true")
+    executorProps.setProperty(THRIFT_LOB_DIRECT_BUFFERS, "true")
+
+    // setup pool properties
+    val props = getAllPoolProperties(jdbcUrl, null,
+      connProperties.poolProps, connProperties.hikariCP, isEmbedded = false)
+    try {
+      // use jdbcUrl as the key since a unique pool is required for each server
+      ConnectionPool.getPoolConnection(jdbcUrl, SnappyStoreClientDialect, props,
+        executorProps, connProperties.hikariCP)
+    } catch {
+      case sqle: SQLException => if (hostList.size == 1 || useLocatorURL) {
+        throw sqle
+      } else {
+        hostList.remove(index)
+        createConnection(connProperties, hostList)
+      }
+    }
+  }
+
+  def connectionProperties(hostList: ArrayBuffer[(String, String)]): ConnectionProperties = {
+
+    // TODO: Check how to make properties Dynamic
+    val map: Map[String, String] = HashMap[String, String](("maxActive", "256"),
+      ("testOnBorrow", "true"), ("maxIdle", "256"), ("validationInterval", "10000"),
+      ("initialSize", "4"), ("driverClassName", "io.snappydata.jdbc.ClientDriver"))
+
+    val poolProperties = new Properties
+    poolProperties.setProperty("driver", "io.snappydata.jdbc.ClientDriver")
+    poolProperties.setProperty("route-query", "false")
+
+    val executorConnProps = new Properties
+    executorConnProps.setProperty("lob-chunk-size", "33554432")
+    executorConnProps.setProperty("driver", "io.snappydata.jdbc.ClientDriver")
+    executorConnProps.setProperty("route-query", "false")
+    executorConnProps.setProperty("lob-direct-buffers", "true")
```

Comment:
Why are all the values hard-coded? Shouldn't these be passed as a property bag to this method which will be obtained from the original create statement options.

Reply:
Temporary removed this piece of code from this PR. As it is specifically required in V2 Connector. Will be using this reference while opening the V2Connector related PR